src/com/amazon/ion/util/Equivalence.java

```diff
@@ -201,6 +204,22 @@ public Builder withEpsilon(double epsilon) {
             return this;
         }
 
+        /**
+         * The maximum depth to which two values will be compared. If the two
+         * values to be compared are still potentially equivalent beyond this
+         * depth, {@link Equivalence#ionValueEquals(IonValue, IonValue)} will
+         * throw an {@link IonException}.
+         * @param maxComparisonDepth the value.
+         * @return this builder.
+         */
+        public Builder withMaxComparisonDepth(int maxComparisonDepth) {
```

Comment:
How do end users configure this? `IonValue.equals()` is where the issue lies, right? I've used that countless times but I've never explicitly built an Equivalence instance. I didn't even know it existed until this PR. 

My assumption was this would be something on `IonSystemBuilder`. At the least we need really clear documentation. Consider adding code comment to where the exception is raised for the poor slub who has to troubleshoot when their whacky (to us) Ion dataset has crazy nested recursion.

Reply:
`IonValue.equals()` calls into the static `Equivalence.ionEquals` method, which uses a builder with the default settings. If users need to use non-default settings, they must create a custom `Equivalence` instance via a builder configured with the desired settings, then call `equivalence.ionValueEquals(value1, value2)` rather than `value1.equals(value2)`. The top-level JavaDoc on the Equivalence class has examples of each technique. I'll also add a note to the exception message directing users to `Equivalence.Builder`.